### ISSUE TITLE
Oddity when importing an Excel file

### ISSUE THREAD

**Issue by rdstern** [africanmathsinitiative/R-Instat#6192]
[MPAT Kenya.xlsx](https://github.com/africanmathsinitiative/R-Instat/files/5899471/MPAT.Kenya.xlsx)

This is a potential file for the R-Instat library.
I import just the household data sheet.
If I do it simply - just press Ok, then it is fine.
But there are -99 values in the data.  If I allow for them, then it still imports fine, but now some columns are designated as character that were numeric before.  This is particularly Qu55 to Qu58.

(To see this either show more variables (Tools > Options) or look at the column metadata.)

Can we specify a missing value code - perhaps multiple, without this happening.  

(This links to the fact that changing from character to numeric in R-Instat is still problematic - long standing limitation!)

**Comment by rdstern** [comment]
@N-thony in the File > Open from File dialogue there is a field called `Missing Value String`.  If that is empty, then importing works fine.  But if I put -99 into that field, then various variables are imported as character and not numeric.

**Comment by N-thony** [comment]
@rdstern I think this is not a problem with R-Instat, it is how to function work. I have tried the code as you can see in the screenshot below and I understood that when you specify a missing value code, the function change automatically the type of columns containing the missing value to a character.
![image](https://user-images.githubusercontent.com/68591383/106583423-bfd18a80-654d-11eb-9461-ac8aa6abb9e4.png)

**Comment by rdstern** [comment]
That's just my problem.  Note that the variable Q55.1 doesn't have any values of -99.  If you don't have the na there in the command, then Q55.1 comes in as numeric.  Why should rio's import change it to become character?  Is there an option to prevent that?

**Comment by N-thony** [comment]
> That's just my problem. Note that the variable Q55.1 doesn't have any values of -99. If you don't have the na there in the command, then Q55.1 comes in as numeric. Why should rio's import change it to become character? Is there an option to prevent that?

Yeah, you are right. If the column doesn't have the missing value code in my case Q55.1 doesn't have a value of -99, rio will still keep then NA and change the type of the column to character. I'm going to look if there is an option to prevent that

**Comment by lilyclements** [comment]
The issue is that there are missing values for some variables in the data set which are given by a blank cell (e.g. for `Q55.1`, but not for `Q7`). These cells, for some reason, become characters when imported into R if the `na` argument is specify.

To fix this you need to run `na = c("-99", "")`

e.g.
```
data2 <- rio::import("C:/Users/user/Downloads/MPAT.Kenya.xlsx", which = 6, na = "-1")
class(data2$Q7)       # numeric variable
class(data2$Q55.1)  # character variable

data3 <- rio::import("C:/Users/user/Downloads/MPAT.Kenya.xlsx", which = 6, na = c("-1", ""))
class(data3$Q55.1)  # numeric variable
```

**Comment by N-thony** [comment]
> c("-99", "")

Thanks @lilyclements , this is very helpfully.
